src/Document.php

```diff
@@ -567,12 +568,16 @@ public function getDownloadLink($params = '', $len = 20)
         $out   = '';
         $open  = '';
         $close = '';
+        $linkedItem = $linkedItem ?? ['tickets_id' => $this->fields['tickets_id']];
         if (
             self::canView()
-            || $this->canViewFile(['tickets_id' => $this->fields['tickets_id']])
+            || $this->canViewFile($linkedItem)
         ) {
+            $fk = array_keys($linkedItem)[0];
+            $items_id = $linkedItem[$fk];
+            $link = "&$fk=$items_id";
```

Comment:
To be more generic, maybe we should always pass link to `itemtype=XXX&items_id=YYY` format.

Reply:
> To be more generic, maybe we should always pass link to itemtype=XXX&items_id=YYY format.

I guess you are talking about the query string of the URL, right ?

Reply:
indeed